suricata/update/main.py

```diff
@@ -424,14 +424,25 @@ def manage_classification(suriconf, files):
 def handle_dataset_files(rule, dep_files):
     if not rule.enabled:
         return
-    load_attr = [el.strip() for el in rule.dataset.split(",") if "load" in el][0]
-    dataset_fname = os.path.basename(load_attr.split(" ")[1])
+    load_attr = [el.strip() for el in rule.dataset.split(",") if "load" in el]
+    state_attr = [el.strip() for el in rule.dataset.split(",") if "state" in el]
+    if load_attr and state_attr:
+        logger.error("Invalid dataset rule")
+        return
+    elif not load_attr and not state_attr:
+        return
+    elif load_attr:
+        ds_attr = load_attr[0]
+    elif state_attr:
+        ds_attr = state_attr[0]
+
+    dataset_fname = os.path.basename(ds_attr.split(" ")[1])
     filename = [fname for fname, content in dep_files.items() if fname == dataset_fname]
     if filename:
         logger.debug("Copying dataset file %s to output directory" % dataset_fname)
         with open(os.path.join(config.get_output_dir(), dataset_fname), "w+") as fp:
             fp.write(dep_files[dataset_fname].decode("utf-8"))
```

Comment:
Something we should address with this PR as well is handle the case where the file doesn't exist in the dataset.

Reply:
What is the right action in that case?
Error out and disable the rule?

Reply:
For now, just a log an error.

Reply:
OK I was just looking at this and I need clarification about the case you are mentioning. We are already checking if the dir contains the filename mentioned with the keyword and only then we try to open it.. Do you mean a case like `load` without any filename given to it?